assets/js/script.js

```diff
@@ -0,0 +1,113 @@
+// Assignment Code
+const generateBtn = document.querySelector("#generate");
+
+// declaring password criteria
+const uppercase = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
+const lowercase = "abcdefghijklmnopqrstuvwxyz";
+const numeric = "0123456789";
+const specialCharacters = " !\"#$%&'()*+,-./:;<=>?@[]^_`{|}~";
+let userPasswordConfirms;
+
+// prompt user password length input and validate
+const getPasswordLength = () => {
+  const passwordLength = prompt(
+    "How many characters do you want your password to have. Please enter between 8 and 128 characters"
+  );
+  const passLengthNum = parseInt(passwordLength, 10);
+  //validate user password length input
+  if (passLengthNum >= 8 && passLengthNum <= 128) {
+    // return the password length
+    return passLengthNum;
+  } else {
+    // alert user to re-try input a correct number
+    alert("Please enter a valid number");
+    return;
+  }
+};
+
+//prompting user to select the password criteria's
+function passwordCriteria() {
```

Comment:
Could replace the function declaration with an es6 arrow function
```suggestion
const passwordCriteria = () => {
```